### PR TITLE
[MDS-5975] - ams regions

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

- Created a `regions` table and added a column in the `project_summary` table for a region id
- Filled the regions table with the names/ids provided by AMS
- Created a new slice to fetch/store/retrieve the region options
- Updated `FacilityOperator.tsx` with new select component utilizing the region options from the new slice
- Updated API to use the new regions table/columns
- Added the region name to the AMS submission
- Refactored the `ProjectSummaryPage.tsx` file to properly dispatch the actions and increase readability/maintainability

[MDS-5975](https://bcmines.atlassian.net/browse/MDS-5975)

image

### CODE REVIEW

File: services/common/src/components/projectSummary/FacilityOperator.tsx [hunk 1]
```diff
@@ -17,10 +17,39 @@ import { getDropdownProvinceOptions } from "@mds/common/redux/selectors/staticCo
...
+
+  const formErrors = useSelector(getFormSyncErrors(FORM.ADD_EDIT_PROJECT_SUMMARY));
+  const {
+    facility_coords_source,
+    zoning,
+    facility_latitude,
+    facility_longitude,
+    legal_land_desc,
+    facility_pid_pin_crown_file_no,
+  } = formValues;
```
Comment (taraepp) [inline services/common/src/components/projectSummary/FacilityOperator.tsx]: I can see where the lat & lon are being used in this file, but are all the other destructured attributes being used?

File: services/core-api/app/api/projects/project_summary/models/project_summary.py [hunk 2]
```diff
@@ -1105,6 +1109,11 @@ def update(self,
         for authorization in authorizations:
             self.create_or_update_authorization(authorization)
 
+        regional_district_name = regional_district_id is not None and Regions.find_by_id(
+            regional_district_id).name or None
+
+        current_app.logger.info(regional_district_name)
```
Comment (taraepp) [inline services/core-api/app/api/projects/project_summary/models/project_summary.py]: probably don't need this log

File: services/common/src/components/projectSummary/FacilityOperator.tsx (hunk 1 above)
Comment (matbusby-fw) [reply services/common/src/components/projectSummary/FacilityOperator.tsx]: Hmmm, good question.  A bunch of this stuff was updated in the rebase.  I'll take a look.

File: services/minespace-web/src/tests/mocks/dataMocks.js [hunk 3]
```diff
@@ -1957,3 +1957,26 @@ export const NOTICE_OF_DEPARTURE_DETAILS = {
     },
   ],
 };
+
+export const REGIONS = [
```
Comment (taraepp) [inline services/minespace-web/src/tests/mocks/dataMocks.js]: There is a bit of a move to dataMocks.tsx in common. Wondering if this should go there.

Comment (matbusby-fw) [reply services/minespace-web/src/tests/mocks/dataMocks.js]: Works for me!

**Review by taraepp — DISMISSED** [review]
LGTM

**Review by henryoforeh-dev — DISMISSED** [review]

**Review by henryoforeh-dev — APPROVED** [review]

**Review by taraepp — APPROVED** [review]

**Pull request merged**